refactor(external-router): remove unnecessary `processErrors` function

Divayang-2006 opened the pull request:
## PR Checklist

- [X] Commit message follows our [contributing guidelines](https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit)
- [ ] Tests added/updated (for bug fixes/features)
- [ ] Documentation added/updated (for bug fixes/features)

## PR Type
<!-- Select the relevant type(s) -->

- [X] Bug fix
- [ ] Feature
- [ ] Style update
- [X] Refactor
- [ ] Test
- [ ] Build
- [ ] CI
- [ ] Docs
- [ ] Performance
- [ ] Other (please describe)

## Current behavior
external-router drivers do not throw errors directly.
Instead, errors are passed to the `daffExternalMatcherTypeGuard` through the `processErrors` function

Fixes: #4107
Part of: # <!-- Keeps issue open -->

## New behavior
implementing direct error handling in the memory, testing, and Magento drivers, 
so that drivers can directly throw errors


## Breaking change?

- [ ] Yes
- [ ] No

<!-- If yes, describe impact and migration path -->

## Additional context
<!-- Any other relevant information -->

Divayang-2006 committed:
refactor(daffio-why-pwa-stats.html): replace DaffRaisedCardComponent with elevated property

Replaced <a daff-raised-card> with <a daff-card [elevated]="true"> for consistency and deprecation cleanup.

Divayang-2006 committed:
Merge branch 'develop' of https://github.com/Divayang-2006/daffodil into my-fix-branch

Divayang-2006 committed:
refactor(external-router): remove unnecessary `processErrors` function

Simplified error handling by removing the redundant `processErrors` function,
and implementing direct error handling in the `ExternalMatcherTypeGuard.ts`.

Divayang-2006 committed:
refactor(external-router): remove unnecessary `processErrors` function
Simplified error handling by removing the redundant `processErrors` function,
and implementing direct error handling in the `ExternalMatcherTypeGuard.ts`.

damienwebdev commented:
@Divayang-2006 In order to merge this PR, you will need to move the same handling that was done by `processErrors` into each of the drivers. Simply deleting the `processErrors` won't work.

Divayang-2006 committed:
fix(external-router-drivers): throw errors for non-success responses #4107

Implement DaffExternalRouter*Error throwing in in-memory, testing, and Magento drivers so failures no longer rely on processErrors.
letting drivers to throw errors directly.

Divayang-2006 committed:
style(external-router-drivers): fix linting issues and formatting

Divayang-2006 commented:
@damienwebdev Sir, 
can you please Re-review the commit !!!

-> i removed `processErrors`.
-> added error handing in driver so that drivers can directly throw errors.

damienwebdev commented on libs/external-router/routing/src/guard/external-matcher-type.guard.ts:
```diff
@@ -36,8 +35,12 @@ export const daffExternalMatcherTypeGuard = (type: string) => (route: DaffRouteW
   return inject(DaffExternalRouterDriver).resolve(daffConvertToPath(segments)).pipe(
-    map(processErrors),
     map(processRedirects),
+    tap(response => {
```
This tap can be removed.

damienwebdev commented on libs/external-router/routing/src/guard/external-matcher-type.guard.ts:
```diff
@@ -64,9 +67,6 @@ export const daffExternalMatcherTypeGuard = (type: string) => (route: DaffRouteW
     //Otherwise something went horribly wrong and we need to bail out.
-    catchError((error) => of(router.parseUrl(config.failedResolutionPath))),
-    catchError(() => of(false)),
+    catchError(() => of(router.parseUrl(config.failedResolutionPath))),
```
This final `catchError(() => of(false)),` should stay in case the router fails to parse `config.failedResolutionPath`

damienwebdev commented:
@griest024 can you give this a once over?

Divayang-2006 committed:
fix(external-router): address review feedback on error handling

Divayang-2006 commented:
@damienwebdev and @griest024 Sir, 
can you please Re-review the Commit !!!

damienwebdev reviewed (approved):
lgtm
